service.go

```diff
@@ -118,11 +118,13 @@ func (s *service) PushLog(ctx context.Context, req *pb.PushLogRequest) (*pb.Push
 	}
 
 	lg := logFromProto(req.Log)
+
+	// Fix concurrency: adding head without having the block guarantee
 	if err := s.threads.store.AddLog(req.ThreadID.ID, lg); err != nil {
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
-	go func() {
+	func() {
 		// Get log records for this new log
 		recs, err := s.getRecords(
 			s.threads.ctx,
```

Comment:
Disabled `go` here, mostly to avoid concurrency.

I think that this `AddLog` that is here is wrong, but I'd like some confirmation/opinion.
If our log is at height 10, and the received log information has head at height 1, then `AddLog` will override the head with a lower height. 

Am I missing something? 

I think the solution is just deleting that code and let `putRecord` do the set head job (as can be seen below in other comment. I'll reference this point in the other comment with ### mark ).

Reply:
Please keep that comment in mind for my next PR that I'm fixing the concurrency problems and deleting this temporary global lock.

I move some things of place since I want to make some clear distinctions between thread-safe methods and not thread-safe. (or must say, goroutine safe). I added comments to methods and that PR will be heavily commented.

I want to refresh what you mention in that PR if possible.